kv/cache_snapshot_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"bytes"
+
+	"github.com/juju/errors"
+	. "github.com/pingcap/check"
+)
+
+var _ = Suite(&testCacheSnapshotSuite{})
+
+type testCacheSnapshotSuite struct {
+	k1, k2, k3, k4 []byte
+	v1, v2, v3, v4 []byte
+	store          MemBuffer
+	cache          Snapshot
+}
+
+func (s *testCacheSnapshotSuite) SetUpTest(c *C) {
+	s.k1 = []byte("k1")
+	s.k2 = []byte("k2")
+	s.k3 = []byte("k3")
+	s.k4 = []byte("k4")
+	s.v1 = []byte("1")
+	s.v2 = []byte("2")
+	s.v3 = []byte("3")
+	s.v4 = []byte("4")
+
+	s.store = NewMemDbBuffer()
+	s.cache = NewCacheSnapshot(&mockSnapshot{s.store})
+}
+
+func (s *testCacheSnapshotSuite) TearDownTest(c *C) {
+	s.cache.Release()
+}
+
+func (s *testCacheSnapshotSuite) TestGet(c *C) {
+	s.store.Set(s.k1, s.v1)
+	s.store.Set(s.k2, s.v2)
+
+	v, err := s.cache.Get(s.k1)
+	c.Assert(err, IsNil)
+	c.Assert(bytes.Compare(v, s.v1), Equals, 0)
+
+	s.store.Set(s.k1, s.v3)
+	v, err = s.cache.Get(s.k1)
+	c.Assert(err, IsNil)
+	c.Assert(bytes.Compare(v, s.v1), Equals, 0)
+
+	s.store.Set(s.k2, s.v4)
+	v, err = s.cache.Get(s.k2)
+	c.Assert(err, IsNil)
+	c.Assert(bytes.Compare(v, s.v4), Equals, 0)
+}
+
+func (s *testCacheSnapshotSuite) TestBatchGet(c *C) {
+	s.store.Set(s.k1, s.v1)
+	s.store.Set(s.k2, s.v2)
+
+	m, err := s.cache.BatchGet([]Key{s.k1, s.k2, s.k3})
+	c.Assert(err, IsNil)
+	c.Assert(bytes.Compare(m[string(s.k1)], s.v1), Equals, 0)
```

Comment:
use DeepEqual or ByteEquals for bytes equal comparison, the same for all followings.